content/blog/dplyr-1-0-0/index.Rmd

```diff
@@ -0,0 +1,116 @@
+    function variants.
+
+-   [Working within rows](/blog/2020/04/dplyr-1-0-0-rowwise/).
+    `rowwise()` has been renewed and revamped to make it easier to perform
+    operations row-by-row. This makes it much easier to solve problems that
+    previously required `lapply()`, `map()`, or friends.
+
+-   [The role of the vctrs package](/blog/2020/04/dplyr-1-0-0-and-vctrs/).
+    dplyr now makes heavy use of [vctrs](http://vctrs.r-lib.org/) behind
+    the scenes. This brings with it greater consistency and (hopefully!)
+    more useful error messages.
+
+-   [Last minute additions](/blog/2020/05/dplyr-1-0-0-last-minute-additions/)
+    `summarise()` now allows you to control how its results are grouped,
+    and there's a new family of functions designed for modifying rows.
+
+You can see the full list of changes in the [release notes](https://github.com/tidyverse/dplyr/releases/tag/v1.0.0).
+
+## New logo
+
+dplyr has a new logo thanks to the talented [Allison Horst](https://allisonhorst.github.io)!
+
+![New dplyr logo](dplyr.png){width=250px}\ 
+
+(Stay tuned for details about how to get this sticker on to your laptop. We have some exciting news coming up!)
+
+## A small teaser
+
+The best way to find out all the cool new features dplyr has to offer is to read through the blog posts linked to above. But thanks to inspiration from [Daniel Anderson](https://twitter.com/datalorax_/status/1258208502960422914) here's one example of fitting two different models to here are bunch of cool features in one single example:
+
+```{r}
+library(dplyr, warn.conflicts = FALSE)
+
+models <- tibble::tribble(
+  ~model_name,    ~ formula,
+  "length-width", Sepal.Length ~ Petal.Width + Petal.Length,
+  "interaction",  Sepal.Length ~ Petal.Width * Petal.Length
+)
+
+iris %>% 
+  nest_by(Species) %>% 
+  left_join(models, by = character()) %>% 
+  rowwise(Species, model_name) %>% 
+  mutate(model = list(lm(formula, data = data))) %>% 
+  summarise(broom::glance(model))
+```
+
+Note the use of:
+
+* The new `nest_by()`, which generate a nested data frame where each row
+  represents one subgroup.
+
+* In `left_join()`, `by = character()` now performs a Cartesian product, 
+  generates every combination of subgroup and model.
+  
+* `rowwise()` and `mutate()` to fit a model to each row.
+
+* The newly powerful `summarise()` to summarise each model with the model
+  fit statistics computed by `broom::glance()`.
+
+## Acknowledgements
+
+dplyr 1.0.0 has been one of the biggest projects that we, as a team, have ever tackled. Almost everyone in the tidyverse team has been involved in some capacity. Special thanks go to Romain François, who in his role as primary developer has been working on this release for over six months, and to Lionel Henry and Davis Vaughn for all their work on the vctrs package. Jim Hester's work on running revdep checks in the cloud also made a big impact on our ability to understand failure modes.
```

Comment:
```suggestion
dplyr 1.0.0 has been one of the biggest projects that we, as a team, have ever tackled. Almost everyone in the tidyverse team has been involved in some capacity. Special thanks goes to Romain François, who in his role as primary developer has been working on this release for over six months, and to Lionel Henry and Davis Vaughn for all their work on the vctrs package. Jim Hester's work on running revdep checks in the cloud also made a big impact on our ability to understand failure modes.
```

Reply:
The plurality of thanks is a hotly debated topic…I think of it as a mass noun with the _A_ being implied. Stet me if you must, but I wanted my stance on the record.

Reply:
Leave it go, since there are multiple thanks in this sentence!

Reply:
A special thanks _goes_ to @ajmcoqui for helping me resolve my thanks angst.